en/flying/terrain_following_holding.md

```diff
@@ -0,0 +1,81 @@
+# Terrain Following/Holding & Range Aid
+
+PX4 supports [Terrain Following](#terrain_following) and [Terrain Hold](#terrain_hold) in [Position](../flight_modes/position_mc.md) and [Altitude modes](../flight_modes/altitude_mc.md), on *multicopters* and *VTOL vehicles in MC mode* that have a [distance sensor](../sensor/rangefinders.md).
+
+PX4 also supports [using a distance sensor as the primary source of altitude data](#distance_sensor_primary_altitude_source) in any mode, either all the time, or just when flying at low altitudes ([Range Aid](#range_aid)).
+
+
+## Terrain Following {#terrain_following}
+
+When *terrain following* is enabled, a vehicle will automatically maintain a relatively constant height above *ground level* if traveling at low altitudes.
+This is useful for avoiding obstacles and for maintaining constant height (e.g. for aerial photography) when flying over varied terrain.
+
+*Terrain following* uses the output of the EKF estimator to provide the altitude estimate, and the estimated terrain altitude (calculated from distance sensor measurements using another estimator) to provide the altitude setpoint.
+As the distance to ground changes, the altitude setpoint adjusts to keep the height above ground constant.
+
+At higher altitudes (when the estimator reports that the distance sensor data is invalid) the vehicle switches to *altitude following*, and will typically fly at a near-constant height above mean sea level (AMSL) using the barometer for altitude data.
```

Comment:
@hamishwillee ...using its primary height sensor for estimating altitude. (This could be baro or GPS, see EKF2_HGT_MODE)

Reply:
Ah, you actually write this below, forget about it then.